From c31a8e6a5d1f782567feeb948c65af42aba2c62b Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Tue, 29 Nov 2016 15:57:08 +0800
Subject: [PATCH 1375/5242] MLK-13607-7 video: mipi_dsi_northwest: add build
 support for this mipi dsi controller

commit  226053ef9c05d33def5383c8d9681ad722aa377b from
https://source.codeaurora.org/external/imx/linux-imx.git

Add kernel build support the NorthWest Logic MIPI DSI module
on the imx7ulp-evk board.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 4f6d2b2e57c357b015b1f88f77b8af6da74d0df6)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/Kconfig  |    9 +++++++--
 drivers/video/fbdev/mxc/Makefile |    1 +
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/video/fbdev/mxc/Kconfig b/drivers/video/fbdev/mxc/Kconfig
index 5641d64..0e3cfeb 100644
--- a/drivers/video/fbdev/mxc/Kconfig
+++ b/drivers/video/fbdev/mxc/Kconfig
@@ -29,10 +29,15 @@ config FB_MXC_MIPI_DSI_SAMSUNG
 	depends on FB_MXC_SYNC_PANEL
 	depends on FB_MXS
 
+config FB_MXC_MIPI_DSI_NORTHWEST
+	tristate "MXC MIPI_DSI_NORTHWEST"
+	depends on FB_MXC_SYNC_PANEL
+	depends on FB_MXS
+
 config FB_MXC_TRULY_WVGA_SYNC_PANEL
 	tristate "TRULY WVGA Panel"
 	depends on FB_MXC_SYNC_PANEL
-	depends on FB_MXC_MIPI_DSI || FB_MXC_MIPI_DSI_SAMSUNG
+	depends on FB_MXC_MIPI_DSI || FB_MXC_MIPI_DSI_SAMSUNG || FB_MXC_MIPI_DSI_NORTHWEST
 
 config FB_MXC_TRULY_PANEL_TFT3P5079E
 	tristate "TRULY Panel TFT3P5079E"
@@ -42,7 +47,7 @@ config FB_MXC_TRULY_PANEL_TFT3P5079E
 config FB_MXC_TRULY_PANEL_TFT3P5581E
 	tristate "TRULY Panel TFT3P5581E"
 	depends on FB_MXC_SYNC_PANEL
-	depends on FB_MXC_MIPI_DSI_SAMSUNG
+	depends on FB_MXC_MIPI_DSI_SAMSUNG || FB_MXC_MIPI_DSI_NORTHWEST
 
 config FB_MXC_LDB
 	tristate "MXC LDB"
diff --git a/drivers/video/fbdev/mxc/Makefile b/drivers/video/fbdev/mxc/Makefile
index dafc050..0056b69 100644
--- a/drivers/video/fbdev/mxc/Makefile
+++ b/drivers/video/fbdev/mxc/Makefile
@@ -1,5 +1,6 @@
 obj-$(CONFIG_FB_MXC_MIPI_DSI)			+= mipi_dsi.o
 obj-$(CONFIG_FB_MXC_MIPI_DSI_SAMSUNG)           += mipi_dsi_samsung.o
+obj-$(CONFIG_FB_MXC_MIPI_DSI_NORTHWEST)         += mipi_dsi_northwest.o
 obj-$(CONFIG_FB_MXC_TRULY_WVGA_SYNC_PANEL)      += mxcfb_hx8369_wvga.o
 obj-$(CONFIG_FB_MXC_TRULY_PANEL_TFT3P5079E)     += mxcfb_otm8018b_wvga.o
 obj-$(CONFIG_FB_MXC_TRULY_PANEL_TFT3P5581E)	+= mxcfb_hx8363_wvga.o
-- 
1.7.9.5

